nion/instrumentation/DriftTracker.py

```diff
@@ -78,6 +81,114 @@ def __drift_changed(self, offset_nm: Geometry.FloatSize, elapsed_time: float) ->
         self.__event_loop.call_soon_threadsafe(functools.partial(self.__update_drift_log_data_item, delta_nm_data))
 
 
+@dataclasses.dataclass
+class DriftResult:
+    """Result of drift calculation returned by DriftCalculator.calculate
+
+    It can interpreted as: Between 'time_window_start' and 'time_window_end' the average drift rate was estimated to
+    be 'drift_rate'.
+
+    'drift_rate' should be in units of nm / s.
+    """
+
+    drift_rate: Geometry.FloatSize
+    time_window_start: datetime.datetime
+    time_window_end: datetime.datetime
+
+
+class DriftCalculator:
+    """Calculate drift rate from images with timestamps
+
+    Used by DriftTracker to perform the actual drift calculation from submitted images.
+
+    There can be different algorithms for drift calculation. Each of them should be implemented in its own class
+    which should be a subclass of this one.
+
+    DriftTracker discovers available drift calculation algorithms from the registry by looking for the component id
+    'drift_calculator'.
+
+    You need to set 'drift_calculator_id' to a unique name that will be used to identify a specifc algorithm.
+    'drift_calculator_name' is optional and will be used as the display name in a UI based drift calculator selection.
+    """
+    drift_calculator_id: str
+    drift_calculator_name: typing.Optional[str] = None
+
+    def calculate(self, first_xdata: typing.Optional[DataAndMetadata.DataAndMetadata], xdata_history: collections.deque) -> typing.Optional[DriftResult]:
```

Comment:
`calculate` -> `calculate_drift_result` I think "calculate" is too general; a more specific method makes it easier to search for in the future.

Reply:
Sure, I can change that.